x/ccv/provider/keeper/distribution.go

```diff
@@ -57,32 +68,225 @@ func (k Keeper) GetAllConsumerRewardDenoms(ctx sdk.Context) (consumerRewardDenom
+			// set the validator commission rate
+			val.Commission.CommissionRates.Rate = cr
 		}
+
+		// allocate the consumer reward tokens to the validator
+		k.distributionKeeper.AllocateTokensToValidator(
+			ctx,
+			val,
+			tokensFraction,
+		)
+
+		// sum the tokens allocated
+		allocated = allocated.Add(tokensFraction...)
+	}
+
+	return allocated
+}
+
+// TransferConsumerRewardsToDistributionModule transfers the rewards allocation of the given consumer chain
+// from the consumer rewards pool to a the distribution module
+func (k Keeper) TransferConsumerRewardsToDistributionModule(
+	ctx sdk.Context,
+	chainID string,
+) (sdk.Coins, error) {
+	// Get coins of the consumer rewards allocation
+	allocation := k.GetConsumerRewardsAllocation(ctx, chainID)
+
+	if allocation.Rewards.IsZero() {
+		return sdk.Coins{}, nil
+	}
+
+	// Truncate coin rewards
+	rewardsToSend, _ := allocation.Rewards.TruncateDecimal()
+
+	// NOTE the consumer rewards allocation isn't a module account, however its coins
+	// are held in the consumer reward pool module account. Thus the consumer
+	// rewards allocation must be reduced separately from the SendCoinsFromModuleToAccount call.
+
+	// Update consumer rewards allocation with the remaining decimal coins
+	allocation.Rewards = allocation.Rewards.Sub(sdk.NewDecCoinsFromCoins(rewardsToSend...))
+
+	// Send coins to distribution module account
+	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ConsumerRewardsPool, distrtypes.ModuleName, rewardsToSend)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	k.SetConsumerRewardsAllocation(ctx, chainID, allocation)
+	return rewardsToSend, nil
+}
+
+// consumer reward pools getter and setter
+
+// GetConsumerRewardsAllocation returns the consumer rewards allocation for the given chain ID
+func (k Keeper) GetConsumerRewardsAllocation(ctx sdk.Context, chainID string) (pool types.ConsumerRewardsAllocation) {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(types.ConsumerRewardsAllocationKey(chainID))
+	k.cdc.MustUnmarshal(b, &pool)
+	return
+}
+
+// SetConsumerRewardsAllocation sets the consumer rewards allocation for the given chain ID
+func (k Keeper) SetConsumerRewardsAllocation(ctx sdk.Context, chainID string, pool types.ConsumerRewardsAllocation) {
+	store := ctx.KVStore(k.storeKey)
+	b := k.cdc.MustMarshal(&pool)
+	store.Set(types.ConsumerRewardsAllocationKey(chainID), b)
+}
+
+// GetConsumerRewardsPool returns the balance
+// of the consumer rewards pool module account
+func (k Keeper) GetConsumerRewardsPool(ctx sdk.Context) sdk.Coins {
+	return k.bankKeeper.GetAllBalances(
+		ctx,
+		k.accountKeeper.GetModuleAccount(ctx, types.ConsumerRewardsPool).GetAddress(),
+	)
+}
+
+// ComputeConsumerTotalVotingPower returns the validator set total voting power
+// for the given consumer chain
+func (k Keeper) ComputeConsumerTotalVotingPower(ctx sdk.Context, chainID string) (totalPower int64) {
```

Comment:
Move function to `validator_set_update.go`

Reply:
Why? It's called in this file, but not in validator_set_update, and it also has nothing to do with the actual valset updates